src/main/java/service/operation/BalanceOperation.java

```diff
@@ -0,0 +1,19 @@
+package service.operation;
+
+import data.db.Storage;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+
+    @Override
+    public void operation(FruitTransaction.Operation operation, FruitTransaction fruitTransaction) {
+        for (Map.Entry<String, Integer> fruit : Storage.fruitsStorage.entrySet()) {
+            if (fruit.getKey().equals(fruitTransaction.getFruit())) {
+                fruit.setValue(fruitTransaction.getQuantity());
+                return;
+            }
+        }
```

Comment:
The use of the 'setValue' method directly on the Map.Entry is not typical and may lead to confusion. It would be better to update the map's value using 'Storage.fruitsStorage.put(key, value)'.